packages/manager/src/managers/SliceMachineManager.ts

```diff
@@ -0,0 +1,362 @@
+import { CustomTypes } from "@prismicio/types-internal";
+import { SharedSliceContent } from "@prismicio/types-internal/lib/content";
+import {
+	SliceMachinePlugin,
+	SliceMachinePluginRunner,
+} from "@slicemachine/plugin-kit";
+
+import { createContentDigest } from "../lib/createContentDigest";
+
+import {
+	PackageChangelog,
+	PackageManager,
+	// PackageVersion,
+	SliceMachineConfig,
+} from "../types";
+import {
+	PrismicAuthManager,
+	PrismicUserProfile,
+} from "../auth/PrismicAuthManager";
+import { createPrismicAuthManager } from "../auth/createPrismicAuthManager";
+
+import { UserManager } from "./user/UserManager";
+import { PrismicRepositoryManager } from "./prismicRepository/PrismicRepositoryManager";
+
+import { PluginsManager } from "./plugins/PluginsManager";
+
+import { ProjectManager } from "./project/ProjectManager";
+import { CustomTypesManager } from "./customTypes/CustomTypesManager";
+import { SlicesManager } from "./slices/SlicesManager";
+import { SnippetsManager } from "./snippets/SnippetsManager";
+import { ScreenshotsManager } from "./screenshots/ScreenshotsManager";
+import { SimulatorManager } from "./simulator/SimulatorManager";
+
+import { VersionsManager } from "./versions/VersionsManager";
+
+import { TelemetryManager } from "./telemetry/TelemetryManager";
+
+type SliceMachineManagerGetStateReturnType = {
+	env: {
+		shortId?: string;
+		intercomHash?: string;
+		manifest: {
+			localSliceSimulatorURL?: string;
+		};
+		repo: string;
+		changelog?: PackageChangelog;
+		packageManager: PackageManager;
+		mockConfig: unknown;
+		framework: unknown; // TODO: Remove
```

Comment:
Currently, no, but we can do one of the following:

1. Track by adapter rather than framework. If a project uses `@slicemachine/adapter-next`, we know they are using Next.js. We can make this work for all adapters automatically since we only need to read the `adapter` property in `slicemachine.config.json`.
2. Detect which framework is being used by reading a project's `package.json`. We would need to write detection code for each framework, which means we wouldn't automatically detect new frameworks like Astro until we explicitly checked for it.

We do some framework detection in `@slicemachine/init` to know which adapter to install.

I think Option #1 is the better approach since it scales to future frameworks without our involvement.